aws-predictions-tensorflow/src/main/java/com/amplifyframework/predictions/tensorflow/TensorFlowPredictionsPlugin.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.predictions.tensorflow;
+
+import android.content.Context;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.WorkerThread;
+
+import com.amplifyframework.AmplifyException;
+import com.amplifyframework.core.Consumer;
+import com.amplifyframework.predictions.PredictionsException;
+import com.amplifyframework.predictions.PredictionsPlugin;
+import com.amplifyframework.predictions.operation.InterpretOperation;
+import com.amplifyframework.predictions.options.InterpretOptions;
+import com.amplifyframework.predictions.result.InterpretResult;
+import com.amplifyframework.predictions.tensorflow.operation.TensorFlowInterpretOperation;
+import com.amplifyframework.predictions.tensorflow.request.TensorFlowTextClassificationRequest;
+import com.amplifyframework.predictions.tensorflow.service.TensorFlowPredictionsService;
+
+import org.json.JSONObject;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * A plugin for Predictions category that uses models from
+ * TensorFlow Lite to carry out tasks offline.
+ */
+public final class TensorFlowPredictionsPlugin extends PredictionsPlugin<TensorFlowPredictionsEscapeHatch> {
+    private static final String TFL_PREDICTIONS_PLUGIN_KEY = "tflPredictionsPlugin";
+
+    private final ExecutorService executorService;
+
+    private TensorFlowPredictionsService predictionsService;
+
+    public TensorFlowPredictionsPlugin() {
+        this.executorService = Executors.newCachedThreadPool();
+    }
+
+    @NonNull
+    @Override
+    public String getPluginKey() {
+        return TFL_PREDICTIONS_PLUGIN_KEY;
+    }
+
+    @Override
+    public void configure(
+            JSONObject pluginConfiguration,
+            @NonNull Context context
+    ) throws AmplifyException {
+        this.predictionsService = new TensorFlowPredictionsService(context);
```

Comment:
As per other PR, you could consider using a `ContextProvider` `@FunctionalInterface` to work around the ordering of object construction, e.g.
```
public final class TensorFlowPredictionsPlugin {
    public TensortFlowPredictionsPlugin() {
        this.predictionsService = new TensorFlowPredictionsService(() -> context);   
    }
}
```
```
final class TensorFlowPredictionsService {
    private final ContextResovler contextResolver;
    TensorFlowPredictionsService(ContextResolver contextResolver) {
        this.contextResovler = contextResovler;
    }

    public void someCategoryBehaviorImpl() {
        // Oh, we need it now ...
        Context resolvedContext = contextResolver.resolveContext();
    }

    @FunctionalItnerface interface ContextResolver {
        Context resolveContext();
    }
}
```